### PR TITLE
[Ch12698] Fix partner import keyerror when 'partner_type_desc' is missing

### PR DESCRIPTION
**Pull request by csabadenes**
[ch12698]

### CODE REVIEW

**Comment by domdinicola** [comment]
what happen if there's no `partner_type` are we still getting a 500?
maybe we can user `_filter_records` to mark the record as invalid

**Comment by csabadenes** [comment]
@domdinicola , as far as i can see, if there's no partner type, the partner is saved as hidden/deleted/blocked and the partner is logged as 'skipped'.

I have noticed another issue.. the data which caused this exception has no 'partner_type_desc' at all, and it should have been filtered by '_filter_records', because 'PARTNER_TYPE_DESC' is mandatory. So in the end there should be no exception.. i'll change the task to 'WIP' and recheck some things.

File: src/etools/applications/partners/views/partner_organization_v2.py [hunk 1]
```diff
@@ -471,6 +471,12 @@ def create(self, request, *args, **kwargs):
 
         country = request.user.profile.country
         partner_sync = PartnerSynchronizer(business_area_code=country.business_area_code)
+
+        partner_resp = partner_sync._filter_records([partner_resp]).pop()
```
Comment (domdinicola) [inline src/etools/applications/partners/views/partner_organization_v2.py]: do we need the `pop` here?

File: src/etools/applications/partners/synchronizers.py [hunk 2]
```diff
@@ -260,8 +260,6 @@ def get_partner_type(partner):
             'GOVERNMENT': 'Government',
             'UN AGENCY': 'UN Agency',
         }
-        if 'PARTNER_TYPE_DESC' in partner:
-            return type_mapping.get(partner['PARTNER_TYPE_DESC'].upper(), None)
```
Comment (domdinicola) [inline src/etools/applications/partners/synchronizers.py]: restore the return

File: src/etools/applications/partners/views/partner_organization_v2.py (hunk 1 above)
Comment (domdinicola) [reply src/etools/applications/partners/views/partner_organization_v2.py]: i would do `if not partner_sync._filter_records([partner_resp])`

Comment (csabadenes) [reply src/etools/applications/partners/views/partner_organization_v2.py]: true, the reassignment was unnecessary and also in some cases wrong, i have fixed it

Comment (domdinicola) [reply src/etools/applications/partners/views/partner_organization_v2.py]: @csabadenes address this

File: src/etools/applications/partners/synchronizers.py (hunk 2 above)
Comment (csabadenes) [reply src/etools/applications/partners/synchronizers.py]: i noticed that mistake bit later, i have restored it.

Comment (csabadenes) [reply src/etools/applications/partners/synchronizers.py]: i have also added few extra tests to check for missing required fields and invalid partner type

File: src/etools/applications/partners/views/partner_organization_v2.py (hunk 1 above)
Comment (csabadenes) [reply src/etools/applications/partners/views/partner_organization_v2.py]: it should be fixed by now

**Pull request merged**